ballerina/sql_client.bal

```diff
@@ -250,9 +281,9 @@ public client class SQLClient {
             }
 
             if key.includes(".") {
-                int splitPosition = <int>key.indexOf(".", 0);
-                string entity = key.substring(0, splitPosition);
-                string fieldName = key.substring(splitPosition + 1, key.length());
+                string[] splits = regex:split(key, "\\.");
```

Comment:
Rather than includes cant we keep a boolean in metadata? To check if this is a field of inner record.

Reply:
Cant we have a map of union type? If we can differentiate between simple type and entity type, I feel like the implementation would  be simpler and easier to read

Reply:
Fixed in [2db76bc](https://github.com/ballerina-platform/module-ballerina-persist/pull/82/commits/2db76bcb4776b1a51558bd4bd1d9b5f5e143213f)

Reply:
Fixed in [7b8b2ef](https://github.com/ballerina-platform/module-ballerina-persist/pull/82/commits/7b8b2efc4881761a41fc05bc0e3ce97b257630fd)